back/engines/commercial/content_builder/app/services/content_builder/layout_service.rb

```diff
@@ -11,5 +11,19 @@ def select_craftjs_elements_for_type(craftjs, type)
     def craftjs_element_of_type?(elt, type)
       elt.is_a?(Hash) && elt['type'].is_a?(Hash) && elt.dig('type', 'resolvedName') == type
     end
+
+    def images(layout)
+      layout_image_codes = []
+
+      layout.craftjs_jsonmultiloc.each_key do |locale|
+        layout.craftjs_jsonmultiloc[locale].each_value do |node|
+          next unless craftjs_element_of_type?(node, 'Image')
+
+          layout_image_codes << node['props']['dataCode']
+        end
+      end
+
+      LayoutImage.where(code: layout_image_codes)
+    end
```

Comment:
I think we will be using this in #3928 & [#4051](https://github.com/CitizenLabDotCo/citizenlab/pull/4051), so it seems a useful method to use in this PR too.